lib/cloud/mocks/aws_sts.go

```diff
@@ -55,17 +56,25 @@ type STSClient struct {
 }
 
 func (m *STSClient) GetCallerIdentity(ctx context.Context, params *sts.GetCallerIdentityInput, optFns ...func(*sts.Options)) (*sts.GetCallerIdentityOutput, error) {
+	if m.Unauth {
+		return nil, trace.AccessDenied("unauthorized")
+	}
```

Comment:
Is it ever possible to get an access denied error from `sts:GetCallerIdentity`? 🤔